ui/tests/acceptance/jobs-list-test.js

```diff
@@ -735,4 +735,477 @@ module('Acceptance | jobs list', function (hooks) {
+        rows = document.querySelectorAll('.job-row');
+        modifyIndexes = Array.from(rows).map((row) =>
+          parseInt(row.getAttribute('data-test-modify-index'))
+        );
+        assert.deepEqual(
+          modifyIndexes,
+          Array(jobsToCreate)
+            .fill()
+            .map((_, i) => i + 1)
+            .reverse()
+            .slice(pageSize, pageSize * 2),
+          'Keynav takes me forward a page'
+        );
+
+        localStorage.removeItem('nomadPageSize');
+      });
+    });
+    module('Live updates are reflected in the list', function () {
+      test('When you have live updates enabled, the list updates when new jobs are created', async function (assert) {
+        localStorage.setItem('nomadPageSize', '10');
+        createJobs(server, 10);
+        await JobsList.visit();
+        assert.dom('.job-row').exists({ count: 10 });
+        let rows = document.querySelectorAll('.job-row');
+        assert.equal(rows.length, 10, 'List is still 10 rows');
+        let modifyIndexes = Array.from(rows).map((row) =>
+          parseInt(row.getAttribute('data-test-modify-index'))
+        );
+        assert.deepEqual(
+          modifyIndexes,
+          Array(10)
+            .fill()
+            .map((_, i) => i + 1)
+            .reverse(),
+          'Jobs are sorted by modify index'
+        );
+        assert.dom('[data-test-pager="next"]').isDisabled();
+
+        // Create a new job
+        server.create('job', {
+          namespaceId: 'default',
+          resourceSpec: Array(1).fill('M: 256, C: 500'),
+          groupAllocCount: 1,
+          modifyIndex: 11,
+          createAllocations: false,
+          shallow: true,
+          name: 'new-job',
+        });
+
+        const controller = this.owner.lookup('controller:jobs.index');
+
+        let currentParams = {
+          per_page: 10,
+        };
+
+        // We have to wait for watchJobIDs to trigger the "dueling query" with watchJobs.
+        // Since we can't await the watchJobs promise, we set a reasonably short timeout
+        // to check the state of the list after the dueling query has completed.
+        await controller.watchJobIDs.perform(currentParams, 0);
+
+        let updatedJob = assert.async(); // watch for this to say "My tests oughta be passing by now"
+        const duelingQueryUpdateTime = 200;
+
+        assert.timeout(500);
+
+        setTimeout(async () => {
+          // Order should now be 11-2
+          rows = document.querySelectorAll('.job-row');
+          modifyIndexes = Array.from(rows).map((row) =>
+            parseInt(row.getAttribute('data-test-modify-index'))
+          );
+          assert.deepEqual(
+            modifyIndexes,
+            Array(10)
+              .fill()
+              .map((_, i) => i + 2)
+              .reverse(),
+            'Jobs are sorted by modify index'
+          );
+
+          // Simulate one of the on-page jobs getting its modify-index bumped. It should remain in place.
```

Comment:
I expected to find more woopsies in all this copypasta, but this here comment is the only one I spotted -- in this test case, the job should not "remain in place" it should get bumped to the top, as asserted down on line 1051 (modify index 5 becomes 12)